### PR TITLE
[HttpClient] Providing additional options to CurlHttpClient section

### PR DESCRIPTION
**Pull request by pizzaminded**
Documentation for symfony/symfony#38026 PR.

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: http_client.rst [hunk 1]
```diff
@@ -728,6 +728,39 @@ When using this component in a full-stack Symfony application, this behavior is
 not configurable and cURL will be used automatically if the cURL PHP extension
 is installed and enabled. Otherwise, the native PHP streams will be used.
 
+Providing additional options to CurlHttpClient
```
Comment (OskarStark) [inline http_client.rst:731]: Additional Options ...


Uppercase letters. Thanks

File: http_client.rst [hunk 2]
```diff
@@ -728,6 +728,39 @@ When using this component in a full-stack Symfony application, this behavior is
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to provide additional cURL options to CurlHttpClient. PHP exposes
```
Comment (OskarStark) [inline http_client.rst:734]: CurlHttpClient in backticks please

File: http_client.rst [hunk 3]
```diff
@@ -728,6 +728,39 @@ When using this component in a full-stack Symfony application, this behavior is
...
+a lot of `cURL options`_ that can be passed to ``curl_setopt`` function, but only some
+of them are used in CurlHttpClient in favor of bigger component portability.
```
Comment (OskarStark) [inline http_client.rst:736]: Same here

File: http_client.rst [hunk 4]
```diff
@@ -728,6 +728,39 @@ When using this component in a full-stack Symfony application, this behavior is
...
+
+    use Symfony\Component\HttpClient\CurlHttpClient;
+
+    $client = new CurlHttpClient();
+
+    $client->request('POST', 'https://...', [
+        // ...
+        'extra' => [
+            'curl' => [
+                \CURLOPT_IPRESOLVE => \CURL_IPRESOLVE_V6
```
Comment (OskarStark) [inline http_client.rst:749]: IMO we can remove the leading slashes to keep the example easier to read

File: http_client.rst (hunk 1 above)
Comment (pizzaminded) [reply http_client.rst]: fixed.

File: http_client.rst (hunk 2 above)
Comment (pizzaminded) [reply http_client.rst]: fixed.

File: http_client.rst (hunk 3 above)
Comment (pizzaminded) [reply http_client.rst]: fixed.

File: http_client.rst (hunk 4 above)
Comment (pizzaminded) [reply http_client.rst]: fixed.

Comment (pizzaminded) [reply http_client.rst]: fixed.

**Comment by javiereguiluz** [comment]
Mikolaj, thanks and congrats on your first Symfony Docs contribution!

It was a fantastic first-time contribution, but we made some minor rewords to make the new contents more similar to the existing contents (but we kept most of your original contribution).

**Pull request merged**